devenv/main.py

```diff
@@ -37,6 +37,19 @@ def initialize_config(config_path: str, defaults: Config) -> None:
     config.read_dict(defaults)
     for section, values in config.items():
         for var, _val in values.items():
+            if section == "devenv" and var == "coderoot":
+                # this is a special case used to make the transition from existing
+                # dev environments easier as we can guess the desired coderoot if
+                # devenv is run inside of a git repo
+                try:
+                    reporoot = gitroot()
+                    coderoot = os.path.abspath(f"{reporoot}/..")
+                    print(f"\nWe autodetected a coderoot: {coderoot}")
+                    config.set(section, var, coderoot)
+                    continue
+                except RuntimeError:
+                    pass
```

Comment:
`try:` / `except:` should wrap the smallest context possible -- and generally be explicit in what exceptions it catches

I think you want:

```python
try:
    reporoot = gitroot()
except RuntimeError:  # TODO: better exception type
    pass
else:
    coderoot = ...
```

Reply:
oh kinda embarassing but TIL about try except else - thanks!

Reply:
oh! it's so good!

Reply:
i don't know how to go about a more specific exception type here because stdout doesn't make it through to  `detail`, and it's not a specific git returncode :(